extensions/mssql/src/dashboard/bookExtensions.ts

```diff
@@ -117,8 +119,18 @@ class AzdataExtensionBookContributionProvider extends Disposable implements Book
 			.map(BookContributions.fromExtension)
 			.reduce(BookContributions.merge, []);
 	}
+
+	private registerCommands(): void {
+		this.contributions.map(book => {
+			let bookName: string = path.basename(book.path);
+			vscode.commands.executeCommand('setContext', bookName, true);
+			vscode.commands.registerCommand('books.' + bookName, async (context) => {
+				vscode.commands.executeCommand('bookTreeView.openBook', book.path, true);
```

Comment:
Why are you making a separate command for each book anyways? Why not just a single command that takes in a book name and then opens the appropriate book?

Reply:
I didn't know we can define commands accepting an argument. Created a an issue to track this:
https://github.com/microsoft/azuredatastudio/issues/6911 
I'll work on that asap.